### ISSUE TITLE
Two RDS Cluster instances are incorrectly put in the same AZ if availablity_zones[] does not match subnet_ids[] 

### ISSUE THREAD

**Issue by brunzefb** [hashicorp/terraform#15236]
### Terraform Configuration Files
```hcl

# Variables
variable "rds_master_username" {
  description = "Aurora RDS username"
  type        = "string"
  default     = "user"
}

variable "rds_master_password" {
  description = "Keycloak Aurora RDS Master Password"
  type        = "string"
  default     = "pw"
}

resource "aws_vpc" "keycloak-vpc" {
  cidr_block = "${var.base_cidr_block}"

  tags {
    Name = "keycloak-vpc"
  }
}

# define public and private subnets
resource "aws_subnet" "public-subnet-keycloak" {
  vpc_id     = "${aws_vpc.keycloak-vpc.id}"
  cidr_block = "${var.subnet_cidr_block}"

  tags {
    Name = "public-subnet-keycloak"
  }
}

resource "aws_subnet" "private-subnet-keycloak" {
  vpc_id            = "${aws_vpc.keycloak-vpc.id}"
  cidr_block        = "${var.private_subnet_cidr_block}"
  availability_zone = "us-east-2c"

  tags {
    Name = "private-subnet-keycloak-2c"
  }
}

resource "aws_subnet" "private-subnet-keycloak2" {
  vpc_id            = "${aws_vpc.keycloak-vpc.id}"
  cidr_block        = "${var.private_subnet_cidr_block2}"
  availability_zone = "us-east-2b"

  tags {
    Name = "private-subnet-keycloak-2b"
  }
}

# the internet gateway is to connect bastion server
resource "aws_internet_gateway" "keycloak-gateway" {
  vpc_id = "${aws_vpc.keycloak-vpc.id}"

  tags {
    Name = "keycloak-gateway"
  }
}

# define route tables
resource "aws_route_table" "keycloak-rt" {
  vpc_id = "${aws_vpc.keycloak-vpc.id}"

  route {
    cidr_block = "0.0.0.0/0"
    gateway_id = "${aws_internet_gateway.keycloak-gateway.id}"
  }

  tags {
    Name = "public-keycloak-rt"
  }
}

resource "aws_route_table" "private-keycloak-rt" {
  vpc_id = "${aws_vpc.keycloak-vpc.id}"

  tags {
    Name = "private-keycloak-rt"
  }
}

# define route table associations
resource "aws_route_table_association" "keycloak" {
  subnet_id      = "${aws_subnet.public-subnet-keycloak.id}"
  route_table_id = "${aws_route_table.keycloak-rt.id}"
}

resource "aws_route_table_association" "private-keycloak" {
  subnet_id      = "${aws_subnet.private-subnet-keycloak.id}"
  route_table_id = "${aws_route_table.private-keycloak-rt.id}"
}

resource "aws_route_table_association" "private-keycloak2" {
  subnet_id      = "${aws_subnet.private-subnet-keycloak2.id}"
  route_table_id = "${aws_route_table.private-keycloak-rt.id}"
}

# Cluster instance, each db instance belongs to a cluster
resource "aws_rds_cluster_instance" "aurora_cluster_instance" {
  count                = 1
  identifier           = "keycloak-aurora-cluster-instance-1"
  cluster_identifier   = "${aws_rds_cluster.aurora_cluster.id}"
  instance_class       = "db.t2.medium"
  db_subnet_group_name = "${aws_db_subnet_group.aurora_subnet_group_keycloak.name}"
  publicly_accessible  = false

  tags {
    Name        = "KeyCloak-Aurora-DB-Instance1"
    ManagedBy   = "terraform"
    Environment = "KeyCloak"
  }

  lifecycle {
    create_before_destroy = true
  }
}

# Cluster instance, each db instance belongs to a cluster
resource "aws_rds_cluster_instance" "aurora_cluster_instance2" {
  count                = 1
  identifier           = "keycloak-aurora-cluster-instance-2"
  cluster_identifier   = "${aws_rds_cluster.aurora_cluster.id}"
  instance_class       = "db.t2.medium"
  db_subnet_group_name = "${aws_db_subnet_group.aurora_subnet_group_keycloak.name}"
  publicly_accessible  = false

  tags {
    Name        = "KeyCloak-Aurora-DB-Instance2"
    ManagedBy   = "terraform"
    Environment = "KeyCloak"
  }

  lifecycle {
    create_before_destroy = true
  }
}

# ISSUE IN HERE!
resource "aws_rds_cluster" "aurora_cluster" {
  cluster_identifier           = "keycloak-aurora-cluster"
  database_name                = "keycloak"
  master_username              = "${var.rds_master_username}"
  master_password              = "${var.rds_master_password}"
  availability_zones           = ["us-east-2b", "us-east-2c"]   # ISSUE HERE ORDER IMPORTANT
  backup_retention_period      = 2
  preferred_backup_window      = "02:00-03:00"
  preferred_maintenance_window = "sat:03:00-sat:04:00"
  db_subnet_group_name         = "${aws_db_subnet_group.aurora_subnet_group_keycloak.name}"
  final_snapshot_identifier    = "keycloak-aurora-cluster"

  vpc_security_group_ids = [
    "${aws_security_group.keycloak_dev_sg.id}",
  ]

  tags {
    Name        = "KeyCloak-Aurora-DB-Cluster"
    ManagedBy   = "terraform"
    Environment = "KeyCloak"
  }

  lifecycle {
    create_before_destroy = true
  }
}

# ISSUE IN HERE
resource "aws_db_subnet_group" "aurora_subnet_group_keycloak" {
  name        = "keycloak_aurora_db_subnet_group"
  description = "Allowed subnets for Aurora DB cluster instances"

  subnet_ids = [
    "${aws_subnet.private-subnet-keycloak2.id}",   # ISSUE IS HERE  this subnet in us-east-2b -- ORDER IS IMPORTANT
    "${aws_subnet.private-subnet-keycloak.id}"     # this subnet in us-east-2c
  ]

  tags {
    Name        = "KeyCloak-Aurora-DB-Subnet-Group"
    ManagedBy   = "terraform"
    Environment = "KeyCloak"
  }
}

```

### Debug Output  -- this is uninteresting -- it succeeds
https://gist.github.com/brunzefb/20d68c4d96186cdd3f84373d7f82f37a

### Expected Behavior
What should have happened?
RDS cluster Instances should have been multi AZ

### Actual Behavior
RDS cluster Instances were in the same AZ  - see below why this happens

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
0. modify script (see important factoids to create an out-of order AZ array
1. `terraform apply`
2. check the created RDS cluster instances -- Multi-AZ will be 'No' and both instances in the same AZ

### Important Factoids
The tf script above does not have the bug  -- to get the bug to show up
```hcl
# In the aws_rds_cluster resource change
 availability_zones           = ["us-east-2b", "us-east-2c"]  to 
 availability_zones           = ["us-east-2c", "us-east-2b"] 
# that will cause the bug to show

# the aws_db_subnet_group defines a list of subnet id's
subnet_ids = [ 
"${aws_subnet.private-subnet-keycloak2.id}",  
    "${aws_subnet.private-subnet-keycloak.id}"     
  ]
# if the availability_zones[] does not match in order the subnet_ids[] the issue shows. up.
# basically the two arrays must match - order is important.
```
Two ways to go about it:
1. Document the issue.
2. Fix Terraform, so that order no longer matters.

Please do either one, as this is pretty obscure when you encounter the issue.  A superficial check of the result may cause you to miss that  you're not running multi-AZ when you actually intended to.
